neuro_extras/image_builder.py

```diff
@@ -0,0 +1,220 @@
+import asyncio
+import base64
+import json
+import logging
+import re
+import uuid
+from dataclasses import dataclass, field
+from typing import Any, AsyncIterator, Dict, Sequence
+
+import click
+from neuromation import api as neuro_api
+from neuromation.api import Preset, Resources
+from neuromation.api.url_utils import normalize_storage_path_uri
+from yarl import URL
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class DockerConfigAuth:
+    registry: str
+    username: str
+    password: str = field(repr=False)
+
+    @property
+    def credentials(self) -> str:
+        return base64.b64encode(f"{self.username}:{self.password}".encode()).decode()
+
+
+@dataclass
+class DockerConfig:
+    auths: Sequence[DockerConfigAuth] = ()
+
+    def to_primitive(self) -> Dict[str, Any]:
+        return {
+            "auths": {auth.registry: {"auth": auth.credentials} for auth in self.auths}
+        }
+
+
+class ImageBuilder:
```

Comment:
Why not in image.py? I thought we create a separate file for each CLI command group.
Or it will be too much?

Reply:
This class is used in `common.py`, '`image.py`, `k8s.py` and `seldon.py` and it seems to be fairly self-contained for being a part of `common.py`, but I'm open to discussion about whether it belongs to `common.py` or not :) wdyt?

Reply:
Oh, I didn't follow imports, my bad 🙈 
This class is pretty large, your solution is the best for me (to keep it in a separate file), thanks for the explanation 👍